bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/handler/channelhelper/CameraChannelHelper.java

```diff
@@ -36,64 +37,72 @@
 public class CameraChannelHelper extends ChannelHelper {
     private static final String QUALITY_CONF_ENTRY = "quality";
     private static final String LIVE_PICTURE = "/live/snapshot_720.jpg";
-    private boolean isLocal;
-    private @Nullable String vpnUrl;
+
+    private @NonNullByDefault({}) String vpnUrl;
     private @Nullable String localUrl;
 
     public CameraChannelHelper(Set<String> providedGroups) {
         super(providedGroups);
     }
 
     public void setUrls(String vpnUrl, @Nullable String localUrl) {
-        this.localUrl = localUrl;
         this.vpnUrl = vpnUrl;
-        this.isLocal = localUrl != null;
-    }
-
-    public @Nullable String getLocalURL() {
-        return localUrl;
+        this.localUrl = localUrl;
     }
 
     @Override
     protected @Nullable State internalGetProperty(String channelId, NAThing naThing, Configuration config) {
         if (naThing instanceof HomeStatusModule camera) {
-            boolean isMonitoring = OnOffType.ON.equals(camera.getMonitoring());
-            switch (channelId) {
-                case CHANNEL_MONITORING:
-                    return camera.getMonitoring();
-                case CHANNEL_SD_CARD:
-                    return toStringType(camera.getSdStatus());
-                case CHANNEL_ALIM_STATUS:
-                    return toStringType(camera.getAlimStatus());
-                case CHANNEL_LIVEPICTURE_VPN_URL:
-                    return toStringType(getLivePictureURL(false, isMonitoring));
-                case CHANNEL_LIVEPICTURE_LOCAL_URL:
-                    return toStringType(getLivePictureURL(true, isMonitoring));
-                case CHANNEL_LIVEPICTURE:
-                    return toRawType(getLivePictureURL(isLocal, isMonitoring));
-                case CHANNEL_LIVESTREAM_VPN_URL:
-                    return getLiveStreamURL(false, (String) config.get(QUALITY_CONF_ENTRY), isMonitoring);
-                case CHANNEL_LIVESTREAM_LOCAL_URL:
-                    return getLiveStreamURL(true, (String) config.get(QUALITY_CONF_ENTRY), isMonitoring);
-            }
+            return switch (channelId) {
+                case CHANNEL_MONITORING -> camera.getMonitoring();
+                case CHANNEL_SD_CARD -> toStringType(camera.getSdStatus());
+                case CHANNEL_ALIM_STATUS -> toStringType(camera.getAlimStatus());
+                default -> liveChannels(channelId, (String) config.get(QUALITY_CONF_ENTRY), camera,
+                        OnOffType.ON.equals(camera.getMonitoring()), localUrl != null);
+            };
         }
         return null;
     }
 
-    private @Nullable String getLivePictureURL(boolean local, boolean isMonitoring) {
-        String url = local ? localUrl : vpnUrl;
-        if (!isMonitoring || (local && !isLocal) || url == null) {
+    public @Nullable String getLivePictureURL(boolean local, boolean isMonitoring) {
+        if (!isMonitoring || (local && (localUrl != null))) {
```

Comment:
Isn't it rather `local && (localUrl == null`

Reply:
That's weird, it is corrected in the third commit..?

Reply:
But in a different class ?